app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js

```diff
@@ -175,6 +175,7 @@ function MwServerControllerFactory($scope, miqService, mwAddDatasourceService, i
   // //////////////////////////////////////////////////////////////////////
 
   $scope.deleteSelectedDr = function() {
+    $document.find('#dr_btn_delete').prop('disabled', true);
```

Comment:
is the actions order here ok? because it first disables the button and only then submits the action. what if the deletion fails? the button will be still disabled, unless I misunderstand the code?

Reply:
I think it's ok, because `submit` is not synchronous and even if it fails we do not know it until we refresh whole page (kinda bummer). For me it does not matter if we first disable the button and then call submit on underlying form.

Reply:
fyi - redux implementation will solve this kind of conflicts :)